spec/requests/notifications_controller_spec.rb

```diff
@@ -228,6 +174,101 @@ def delete_notification(resp_code, matcher)
               read_high_priority.id
             ])
           end
+
+          it "should not bump last seen reviewable in readonly mode" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            Discourse.received_redis_readonly!
+            expect {
+              get "/notifications.json", params: { recent: true }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          ensure
+            Discourse.clear_redis_readonly!
+          end
+
+          it "should not bump last seen reviewable if the user can't seen reviewables" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: { recent: true, bump_last_seen_reviewable: true }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          end
+
+          it "should not bump last seen reviewable if the silent param is present" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: {
+                recent: true,
+                silent: true,
+                bump_last_seen_reviewable: true
+              }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          end
+
+          it "should not bump last seen reviewable if the bump_last_seen_reviewable param is not present" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: { recent: true, silent: true }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          end
+
+          it "bumps last_seen_reviewable_id" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            expect(user.last_seen_reviewable_id).to eq(nil)
+            reviewable = Fabricate(:reviewable)
+            get "/notifications.json", params: { recent: true, bump_last_seen_reviewable: true }
+            expect(user.reload.last_seen_reviewable_id).to eq(reviewable.id)
+
+            reviewable2 = Fabricate(:reviewable)
+            get "/notifications.json", params: { recent: true, bump_last_seen_reviewable: true }
+            expect(user.reload.last_seen_reviewable_id).to eq(reviewable2.id)
+          end
+
+          it "includes pending reviewables when the setting is enabled" do
```

Comment:
I think it'll be clearer if we can specify which setting is enabled.

Reply:
the setting is specified in the `context` block that wraps this test. Maybe I should move this outside that `context` block? 🤔